tests/exporters/common/test_tasks_manager.py

```diff
@@ -63,3 +66,91 @@ def test_all_onnx_models_are_registered(self):
 
     def test_all_tflite_models_are_registered(self):
         return self._check_all_models_are_registered("tflite", "TFLiteConfig")
+
+    def test_register(self):
+        # Case 1: We try to register a config that was already registered, it should not register anything.
+        register_for_onnx = TasksManager.create_register("onnx")
+
+        @register_for_onnx("bert", "sequence-classification")
+        class BadBertOnnxConfig(BertOnnxConfig):
+            pass
+
+        bert_config_constructor = TasksManager.get_exporter_config_constructor(
+            "onnx",
+            model_type="bert",
+            task="sequence-classification",
+        )
+        bert_onnx_config = bert_config_constructor(BertConfig())
+
+        self.assertNotEqual(
+            bert_onnx_config.__class__,
+            BadBertOnnxConfig,
+            "Registering an already existing config constructor should not do anything unless overwrite_existing=True.",
+        )
+
+        # Case 2: We try to register a config that was already registered, but authorize overwriting, it should register
+        # the new config.
+        register_for_onnx = TasksManager.create_register("onnx", overwrite_existing=True)
+
+        @register_for_onnx("bert", "sequence-classification")
+        class BadBertOnnxConfig2(BertOnnxConfig):
+            pass
+
+        bert_config_constructor = TasksManager.get_exporter_config_constructor(
+            "onnx",
+            model_type="bert",
+            task="sequence-classification",
+        )
+        bert_onnx_config = bert_config_constructor(BertConfig())
+
+        self.assertEqual(
+            bert_onnx_config.__class__,
+            BadBertOnnxConfig2,
+            (
+                "Registering an already existing config constructor with overwrite_existing=True should overwrite the "
+                "old config constructor."
+            ),
+        )
+
+        # Case 3: Registering an unknown task.
+        with self.assertRaisesRegex(ValueError, "The TasksManager does not know the task called"):
+
+            @register_for_onnx("bert", "this is a wrong name for a task")
+            class UnknownTask(BertOnnxConfig):
+                pass
+
+        # Case 4: Registering for a new backend.
```

Comment:
For unknown backend, will that really work as it is not validated by `is_backend_available`:
https://github.com/huggingface/optimum/blob/7ecc5cac5dd16e65bfdb63e54a8751fc3ac522f7/optimum/exporters/tasks.py#L49-L54

Reply:
For a custom backend we cannot check for sure. But since to register it you need to import the classes, it actually means that the subpackage is installed.